static/app/views/traces/content.tsx

```diff
@@ -79,19 +79,17 @@ function Content() {
   const handleClearSearch = useCallback(
     (searchIndex: number) => {
       const newQueries = [...queries];
-      if (typeof newQueries[searchIndex] !== undefined) {
-        delete newQueries[searchIndex];
-        browserHistory.push({
-          ...location,
-          query: {
-            ...location.query,
-            cursor: undefined,
-            query: newQueries,
-          },
-        });
-        return true;
-      }
-      return false;
+      // TODO: do we need to return false when `newQueries[searchIndex] === undefined`?
```

Comment:
I didn't want to 'fix' the existing if-statement because that would cause a change in behavior. But if we need that guard to exist, then it only needs to be `if (newQueries[searchIndex] === undefined) { ... `